vault/client.go

```diff
@@ -28,30 +28,51 @@ type AuthStrategy interface {
 	Revokable() bool
 }
 
+// GetenvFunc returns the supplied environment variable value
+type GetenvFunc func(string) string
```

Comment:
Admittedly... injecting a `GetenvFunc` was a _long_ way to go to avoid having to monkey with the environment variables in the tests. I went down this rabbit hole largely because I have unsetting/setting env vars in unit tests... it doesn't give me warm fuzzies.